### ISSUE TITLE
ANTs registration initial_moving_transform requires invert_initial_moving_transform

### ISSUE THREAD

**Issue by ashgillman** [nipy/nipype#1569]
When I run `Registration` using `intial_moving_transform`, I get an error:

```
[160808-21:37:39,795 workflow INFO:
	 Executing node _registration_lvl21 in dir: /home/ash/proj/pylx/align/registration_lvl2/mapflow/_registration_lvl21
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/pipeline/engine/nodes.py", line 392, in run
    self._run_interface()
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/pipeline/engine/nodes.py", line 502, in _run_interface
    self._result = self._run_command(execute)
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/pipeline/engine/nodes.py", line 618, in _run_command
    cmd = self._interface.cmdline
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/interfaces/base.py", line 1415, in cmdline
    allargs = self._parse_inputs()
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/interfaces/base.py", line 1650, in _parse_inputs
    arg = self._format_arg(name, spec, value)
  File "/nix/store/jn78f3vv1w96xlhhq0ps3n0xsfiizxv0-python3.4-nipype-0.12.0-rc1/lib/python3.4/site-packages/nipype/interfaces/ants/registration.py", line 816, in _format_arg
    do_invert_transform = int(self.inputs.invert_initial_moving_transform)
TypeError: int() argument must be a string, a bytes-like object or a number, not '_Undefined']
```

I'm guessing the easiest solution is to specify `usedefault=True` for the `invert_initial_moving_transform` trait. Alternatively, a check could be done around the offending line. If you'd like, I can make a PR.

**Comment by blakedewey** [comment]
There is a check for this in the argument format, but it only has an exception for ValueError. We could also just extend this to have TypeError or check for isdefined.

**Comment by fcoldrain** [comment]
I have this problem too. Currently resolved by adding a line:
antsregistration.inputs.invert_initial_moving_transform = False (or True depending on aim)

However, after specifying this line, another red colored warning will show when running the workflow:

UserWarning: Input invert_initial_moving_transform requires inputs: initial_moving_transform
warn(msg)

In my code the initial_moving_transform has already been specified so I have no idea why nipype warn me this. Fortunately the warning does not stop the workflow from being done. 

Is a fix for this issue possible?

**Comment by blakedewey** [comment]
The error should be fixed with #1577. The warning is put in place to remind users to set both inputs, as they are used together. This warning does not take into account runtime assignments (using wf.connect). We could remove this, as we have a check for default, thoughts?

**Comment by ashgillman** [comment]
Looks like #1577 fixes this, so I am closing
